src/imperative/imperative.cc

```diff
@@ -232,16 +237,13 @@ void Imperative::RecordOp(
     node->inputs.resize(inputs.size());
   }
 
-  std::vector<bool>& save_inputs = *p_save_inputs;
```

Comment:
I feel using these reference is better for readability. Also it's consistent with line https://github.com/apache/incubator-mxnet/pull/14836/files#diff-2d0bd6acfa276757ae59106dbed8e3e5R158

Reply:
I don't think is consistent with requiring the use of pointers for output arguments, if you then change it to a reference there's no point on passing pointers in the first place. Please take that into consideration.

Reply:
Please agree with @anirudhacharya on what you guys think is best. I think it comes down to "developer preference" as Tianqi would say. Your comments request in different directions.